nix/README.md

```diff
@@ -0,0 +1,62 @@
+# Nix
+
+The Nix package derivation is currently defined for all [default systems](https://github.com/nix-systems/default), and it can be integrated into NixOS and Home-Manager configurations as below.
+
+## Installation
+
+Here's how to reference the package derivation (and explicitly pin it) in your `flake.nix`:
+
+``` nix
+{
+  inputs.gauntlet.url = github:project-gauntlet/gauntlet/<gauntlet_version_repository_tag>;
+  inputs.gauntlet.inputs.nixpkgs.follows = "nixpkgs";
+}
+```
+
+The package lives can be referenced directly with `gauntlet.packages.${system}.default` or integrated with the overlay `gauntlet.overlays.default`.
+
+## Configuration
+
+Under `programs.gauntlet`, the options provide the following:
+
+1. `enable`: adds executable to system path
+2. `service.enable`: runs daemon with systemd (MacOS launchd not yet supported)
+
+The examples below assume flake inputs are passed to `nixpkgs.lib.nixosSystem` and `home-manager.lib.mkHomeManagerConfiguration` respectively as `inputs` parameter.
+
+### NixOS
+
+``` nix
+{inputs, ...}: {
+  imports = [inputs.gauntlet.nixosModules.default];
+  programs.gauntlet = {
+    enable = true;
+    service.enable = true;
+  };
+}
+```
+
+### Home-Manager
+
+Once `config.toml` is [supported](../README.md#application-config), Home-Manager can populate its contents with `programs.gauntlet.config`.
+
+``` nix
+{inputs, ...}: {
+  imports = [inputs.gauntlet.homeManagerModules.default];
+  programs.gauntlet = {
+    enable = true;
+    service.enable = true;
+    config = {};
+  };
+}
+```
+
+## Development
+
+Because building `librusty_v8` takes forever, we follow nixpkgs precedent and fetch binaries in a fixed-output-derivation. There is a unique output hash per system, so for convenience, using `nix run .#fetch-rusty-v8-hashes` will show how to replace `RUSTY_V8_ARCHIVE`.
+
+Due to the npm `git://` dependency for `@project-gauntlet/tools`, we must also use `fetchNpmDeps` with a hash instead of `importNpmLock`. Whenever `package-lock.json` is updated, this must be too.
```

Comment:
Can you please, write more detailed description of what exactly needs to be updated (and how) when dependencies are updated and when new version is released with respective tag?